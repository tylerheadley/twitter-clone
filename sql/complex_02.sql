/*
 * Like the query in 01.sql that calculates the hashtags that are commonly used with the hashtag #coronavirus,
 * but also returns the total number of times each hashtag is used.
 */
WITH tweet_tags_distinct AS (
    SELECT DISTINCT * FROM tweet_tags
)
SELECT t1.tag, shared_count, total_count
FROM (
    SELECT lower(t1.tag) as tag,count(*) as shared_count
    FROM tweet_tags_distinct t1
    INNER JOIN tweet_tags_distinct t2 ON t1.id_tweets = t2.id_tweets
    WHERE
        lower(t2.tag)='#coronavirus'
    GROUP BY (1)
    LIMIT 100
) t1
INNER JOIN (
    SELECT 
        lower(tag) as tag,
        count(*) as total_count
        FROM tweet_tags_distinct
        GROUP BY (1)
    ) t3 ON t1.tag = t3.tag
ORDER BY shared_count DESC,tag
;
