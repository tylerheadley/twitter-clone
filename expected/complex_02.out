          tag          | shared_count | total_count 
-----------------------+--------------+-------------
 #coronavirus          |           15 |          15
 #covid19              |            5 |          60
 #argentina            |            3 |           7
 #lockdown             |            2 |          16
 #wearamask            |            2 |           4
 #2021makeawish        |            1 |           9
 #auguri2021           |            1 |           1
 #buonanno2021         |            1 |           2
 #cov                  |            1 |           1
 #covid_19             |            1 |          10
 #covid19italia        |            1 |           1
 #cuidémonosentretodos |            1 |           1
 #distancelearning     |            1 |           1
 #fernandosimón        |            1 |           1
 #florida              |            1 |          10
 #goodbye2020          |            1 |          23
 #hello2021            |            1 |          34
 #ignaciocamacho       |            1 |           1
 #lockdownitalia       |            1 |           1
 #outbreak             |            1 |           1
 #pandemic             |            1 |           2
 #plancoronavirus      |            1 |           1
 #quarantine           |            1 |           2
 #socialdistanacing    |            1 |           1
 #stayhome             |            1 |           3
 #temaestinzione       |            1 |           1
 #vaccinare24h         |            1 |           1
 #vaccinazioni         |            1 |           1
 #vaccincovid          |            1 |           2
 #vaccine              |            1 |           6
 #vaccino              |            1 |           1
(31 rows)

